### PR TITLE
JAMES-2991 Bind ComputeMessageFastViewProjectionListener in JMAP prod…

### PR DESCRIPTION
**Pull request by chibenwa**
…ucts

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[24778913926c366bfb052f1160661d4391704b30] [ERROR] Failures: 
[24778913926c366bfb052f1160661d4391704b30] [ERROR]   CassandraGetMessagesMethodTest 
[24778913926c366bfb052f1160661d4391704b30] Expecting:
[24778913926c366bfb052f1160661d4391704b30]   <"(Empty)">
[24778913926c366bfb052f1160661d4391704b30] to be equal to:
[24778913926c366bfb052f1160661d4391704b30]   <"testmail">
[24778913926c366bfb052f1160661d4391704b30] ignoring whitespace differences
```

File: server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessageFastViewProjectionListener.java [hunk 1]
```diff
@@ -95,11 +91,6 @@ private void handleAddedEvent(Added addedEvent, MailboxSession session) throws M
 
     @VisibleForTesting
     MessageFastViewPrecomputedProperties computeFastViewPrecomputedProperties(MessageResult messageResult) throws MailboxException, IOException {
-        MessageFullView message = messageFullViewFactory.fromMessageResults(ImmutableList.of(messageResult));
-
-        return MessageFastViewPrecomputedProperties.builder()
-            .preview(Preview.from(message.getPreview().getValue()))
-            .hasAttachment(message.isHasAttachment())
-            .build();
+        return messageFastViewPrecomputedPropertiesFactory.from(messageResult);
```
Comment (chibenwa) [inline server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessageFastViewProjectionListener.java]: Not relying on `messageFullViewFactory.fromMessageResults` allows to be purely CPU bound (mime parsing over a byte array) hence no issue scheduling this on the parrallel scheduler...

(That is why I cherry-picked this comment)

**Comment by chibenwa** [comment]
test this please

File: server/data/data-jmap/src/main/java/org/apache/james/jmap/api/projections/MessageFastViewPrecomputedProperties.java [hunk 2]
```diff
@@ -66,6 +73,27 @@ public MessageFastViewPrecomputedProperties build() {
...
+        }
+
+        public MessageFastViewPrecomputedProperties from(MessageResult messageResult) throws MailboxException, IOException {
+            return builder()
+                .preview(previewFactory.fromMessageResult(messageResult))
+                .hasAttachment(hasAttachment(messageResult.getLoadedAttachments()))
+                .build();
+        }
+
+        private boolean hasAttachment(List<MessageAttachment> attachments) {
```
Comment (remk) [inline server/data/data-jmap/src/main/java/org/apache/james/jmap/api/projections/MessageFastViewPrecomputedProperties.java]: maybe `hasNotInlineAttachment` ? or a better worded version of that ?

Comment (chibenwa) [reply server/data/data-jmap/src/main/java/org/apache/james/jmap/api/projections/MessageFastViewPrecomputedProperties.java]: I don't like negative wording.

I would prefer `hasAttachmentWithContentDispositionAttachment` if we want to improve this point...

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**